### PR TITLE
[Issue-721][tabular] IdentifierLeakage fix

### LINKED ISSUE #721: [BUG] Identifier Leakage is using deprecated columns for datetime and index
Identifier Leakage is using `datetime_name` and `index_name` which may exist, but index and datetime may also exist on index columns instead.

### PR DESCRIPTION
**Pull request by yromanyshyn**
resolves #721

### CODE REVIEW

**Comment by matanper** [comment]
I don't think this fix is correct, I think you need to use the index and date properties on the dataset. But I never touched those properties so better ask also @noamzbr

File: deepchecks/tabular/checks/methodology/identifier_leakage.py [hunk 1]
```diff
@@ -58,15 +59,24 @@ def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResul
         dataset.assert_label()
         label_name = dataset.label_name
 
-        relevant_columns = list(filter(None, [dataset.datetime_name, dataset.index_name, label_name]))
+        relevant_data = pd.DataFrame({
+            it.name: it
```
[the remark this answers is not in the document]
Comment (yromanyshyn) [reply deepchecks/tabular/checks/methodology/identifier_leakage.py:63]: > name is copied why should not it keep correct name?

index_col code
```python
return pd.Series(self.data.index.get_level_values(0), name=self.data.index.name, index=self.data.index)
```

File: tests/checks/methodology/identifier_leakage_test.py [hunk 2]
```diff
@@ -29,13 +29,33 @@ def util_generate_dataframe_and_expected():
...
+        assert_that(value, close_to(expected[key], 0.1))
+
+
+def test_identifier_leakage_with_extracted_from_dataframe_datatime_index():
+    df, expected = generate_dataframe_and_expected()
+    df.set_index('x2', inplace=True)
+    dataset = Dataset(df=df, label='label', set_datetime_from_dataframe_index=True)
+    result = IdentifierLeakage().run(dataset=dataset)
+    for key, value in result.value.items():
+        assert_that(key, is_in(expected.keys()))
```
Comment (noamzbr) [inline tests/checks/methodology/identifier_leakage_test.py:55]: Does this ensure that the name of the datetime column identified by IdentifierLeakage is indeed x2?

File: deepchecks/tabular/checks/methodology/identifier_leakage.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (noamzbr) [reply deepchecks/tabular/checks/methodology/identifier_leakage.py:63]: Right, and the code in `get_datetime_column_from_index` sets the name to "datetime", so I guess we're good.

**Review by noamzbr — APPROVED** [review]

**Review by matanper — APPROVED** [review]

**Pull request merged**